audformat/core/utils.py

```diff
@@ -1265,6 +1267,60 @@ def union(
     if len(set(types)) != 1:
         objs = [to_segmented_index(obj) for obj in objs]
 
+    return union_misc(objs)
+
+
+def union_misc(
+    objs: typing.Sequence[pd.Index],
+) -> pd.Index:
+    r"""Create union of index objects.
+
+    Requires that levels and dtypes
+    of all objects match,
+    see :func:`audformat.utils.is_index_alike`.
+    Unlike :func:`audformat.utils.union`
+    index objects must not be conform to
+    :ref:`table specifications <data-tables:Tables>`.
+
+    Args:
+        objs: index objects
+
+    Returns:
+        union of index objects
+
+    Raises:
+        ValueError: if level and dtypes of objects do not match
+
+    Example:
+        >>> index1 = pd.Index([0, 1], name='idx')
+        >>> index2 = pd.Index([1, 2], dtype='Int64', name='idx')
+        >>> union_misc([index1, index2])
+        Index([0, 1, 2], dtype='Int64', name='idx')
+        >>> index3 = pd.MultiIndex.from_arrays(
+        ...     [['a', 'b', 'c'], [0, 1, 2]],
+        ...     names=['idx1', 'idx2'],
+        ... )
+        >>> index4 = pd.MultiIndex.from_arrays(
+        ...     [['b', 'c'], [1, 3]],
+        ...     names=['idx1', 'idx2'],
+        ... )
+        >>> union_misc([index3, index4])
+        MultiIndex([('a', 0),
+                    ('b', 1),
+                    ('c', 2),
+                    ('c', 3)],
+                   names=['idx1', 'idx2'])
```

Comment:
I would skip the definition of the index entries as it makes it easier to compare them.
Hence, I would update the examples to:

```python
        >>> union(
        ...     [
        ...         pd.Index([0, 1], name='idx'),
        ...         pd.Index([1, 2], dtype='Int64', name='idx'),
        ...     ]
        ... )
        Index([0, 1, 2], dtype='Int64', name='idx')
        >>> union(
        ...     [
        ...         pd.MultiIndex.from_arrays(
        ...             [['a', 'b', 'c'], [0, 1, 2]],
        ...             names=['idx1', 'idx2'],
        ...         ),
        ...         pd.MultiIndex.from_arrays(
        ...             [['b', 'c'], [1, 3]],
        ...             names=['idx1', 'idx2'],
        ...         ),
        ...    ]
        ... )
        MultiIndex([('a', 0),
                    ('b', 1),
                    ('c', 2),
                    ('c', 3)],
                   names=['idx1', 'idx2'])
        >>> union(
        ...     [
        ...         pd.Index(['spk1'], name='speaker'),
        ...         pd.MultiIndex.from_arrays([['spk2']], names=['speaker']),
        ...     ]
        ... )
        Index(['spk1', 'spk2'], dtype='object', name='speaker')
```

To be in line I would further simplify the docstring examples of `audformat.utils.union()` to:

```python
        >>> union(
        ...     [
        ...         filewise_index(['f1', 'f2', 'f3']),
        ...         filewise_index(['f2', 'f3', 'f4']),
        ...     ]
        ... )
        Index(['f1', 'f2', 'f3', 'f4'], dtype='object', name='file')
        >>> union(
        ...     [
        ...         segmented_index(['f2'], [0], [1]),
        ...         segmented_index(['f1', 'f2'], [0, 1], [1, 2]),
        ...     ]
        ... )
        MultiIndex([('f2', '0 days 00:00:00', '0 days 00:00:01'),
                    ('f1', '0 days 00:00:00', '0 days 00:00:01'),
                    ('f2', '0 days 00:00:01', '0 days 00:00:02')],
                   names=['file', 'start', 'end'])
        >>> union(
        ...     [
        ...         filewise_index(['f1', 'f2']),
        ...         segmented_index(['f1', 'f2'], [0, 0], [1, 1]),
        ...     ]
        ... )
        MultiIndex([('f1', '0 days',               NaT),
                    ('f2', '0 days',               NaT),
                    ('f1', '0 days', '0 days 00:00:01'),
                    ('f2', '0 days', '0 days 00:00:01')],
                   names=['file', 'start', 'end'])
```

Reply:
done